go/src/models/requirement.go

```diff
@@ -4,9 +4,9 @@ type Requirement struct {
 	ID               uint   `gorm:"primaryKey"`
 	Name             string `json:"reqname"`
 	Description      string `json:"desc"`
-	CoursesPermitted []uint `json:"coursesPermitted"` // id of courses that can be used for this requirement
-	CoursesExcepted  []uint `json:"coursesExcepted"`  // id of courses that cannot be used for this requirement
+	CoursesPermitted []Course `gorm:"foreignKey:ID"` // id of courses that can be used for this requirement
+	CoursesExcepted  []Course `gorm:"foreignKey:ID"` // id of courses that cannot be used for this requirement
 	Year             uint   `json:"year"`             // what year the requirement needs to be met e.g. 1 for year 1, ..., 5 for grad
 	CreditsRequired  uint   `json:"creditsRequired"`  // required # of credits from courses permitted
-	Alternatives     []uint `json:"alternatives"`
+	Alternatives     []Course `gorm:"foreignKey:ID"`
```

Comment:
Not too sure what "Alternatives" represents here. If it's not courses, let me know so I can change it.

Reply:
list of courses makes sense! ty for making this change